expander.go

```diff
@@ -67,14 +62,33 @@ func initResolutionCache() ResolutionCache {
 	}}
 }
 
+// resolverContext allows to share a context during spec processing.
+// At the moment, it just holds the index of circular references found.
+type resolverContext struct {
+	// circulars holds all visited circular references, which allows shortcuts.
+	// NOTE: this is not just a performance improvement: it is required to figure out
```

Comment:
I removed your lock here. Please advise. I don't understand how come we could have a concurrency issue here.
On my side I ran the validate tests without problem.

Reply:
I just tried to re run the tests too, and couldn't reproduce it. I swear it was happening last night, so strange.